docs/tools/flow-cli/migrate/migrate-emulator-state.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Migrating Emulator state to Cadence 1.0
+description: How to locally test storage migration and staged contract upgrades on Emulator
+sidebar_position: 6
+---
+
+Flow CLI provides a command `flow migrate state` to migrate your local emulator
+state to Cadence 1.0, and to deploy upgraded contracts to the emulator.
+This is useful for locally testing the staged contract upgrades, and to see how the
+new network and the upgraded contracts would behave with the migrated data.
+
+## Migration Guide
+
+To test the migration, first you would need an emulator state, created with an emulator/CLI release
+that uses a pre-1.0 Cadence version.
+
+### Getting the old state
+
+It is recommended to use the latest pre-1.0 CLI version (e.g: `v1.14.*`) for creating the old state.
+This is because the emulator shipped with older CLI versions may not include some of the system contracts
+(e.g: `RandomBeaconHistory` contract) that are included by default in the newer versions of the emulator/CLI.
+Such contracts are required for the migration.
+
+To get an emulator state with Flow CLI `1.14.0`:
+
+- Start the emulator with the `--persist` flag.
+  ```shell
+  flow emulator --persist
+  ```
+
+- Deploy the pre-1.0 project/contracts to the emulator
+- Run transactions if there are any.
+- Stop the emulator (Ctrl-C on *nix platforms, and Ctrl-Break or Ctrl-Pause on Windows).
+  It is important to make sure the emulator is stopped before taking the snapshot,
+  so that any pending in-memory data would be written to the persisted state properly.
```

Comment:
```suggestion
  **It is important to make sure the emulator is stopped before taking the snapshot,
  so that any pending in-memory data would be written to the persisted state properly.**
```